src/common/snippets/src/lowered_expr.cpp

```diff
@@ -0,0 +1,556 @@
+    }
+    auto loop_ids = (*loop_end_pos)->get_loop_ids();
+    while (std::find(loop_ids.begin(), loop_ids.end(), loop_id) != loop_ids.end()) {
+        loop_end_pos = std::next(loop_end_pos);
+        loop_ids = (*loop_end_pos)->get_loop_ids();
+    }
+}
+
+void LoweredLoopManager::get_io_loop_ports(LoweredExprIR& linear_ir,
+                                           LoweredExprIR::constExprIt loop_begin_pos,
+                                           LoweredExprIR::constExprIt loop_end_pos,
+                                           std::vector<LoweredExprPort>& entries,
+                                           std::vector<LoweredExprPort>& exits) {
+    std::vector<LoweredExprPtr> body_exprs;
+    for (auto expr_it = loop_begin_pos; expr_it != loop_end_pos; ++expr_it) {
+        body_exprs.push_back(*expr_it);
+    }
+    get_io_loop_ports(linear_ir, body_exprs, entries, exits);
+}
+
+void LoweredLoopManager::get_io_loop_ports(LoweredExprIR& linear_ir,
+                                           const std::vector<LoweredExprPtr>& body_exprs,
+                                           std::vector<LoweredExprPort>& entries,
+                                           std::vector<LoweredExprPort>& exits) {
+    entries.clear();
+    exits.clear();
+    for (size_t i = 0; i < body_exprs.size(); ++i) {
+        auto expr = body_exprs[i];
+        const auto inputs = expr->get_inputs();
+        const auto outputs = expr->get_outputs();
+
+        for (size_t in_port = 0; in_port < inputs.size(); ++in_port) {
+            const auto in_td = inputs[in_port];
+            const auto parent_expr = linear_ir.get_expr_by_output(in_td);
+            if (!ov::is_type<opset1::Constant>(parent_expr->get_node()) &&
+                std::find(body_exprs.begin(), body_exprs.begin() + i, parent_expr) == body_exprs.begin() + i) {
+                entries.push_back({expr, in_port});
+            }
+        }
+
+        for (size_t out_port = 0; out_port < outputs.size(); ++out_port) {
+            const auto out_td = outputs[out_port];
+            const auto consumer_exprs = linear_ir.get_exprs_by_input(out_td);
+            for (const auto& conumer_expr : consumer_exprs) {
+                if (std::find(body_exprs.begin() + i, body_exprs.end(), conumer_expr) == body_exprs.end()) {
+                    exits.push_back({expr, out_port});
+                    break;
+                }
+            }
+        }
+    }
+}
+
+void LoweredLoopManager::marking(LoweredExprIR& linear_ir,
+                                 LoweredExprIR::constExprIt loop_begin_pos, LoweredExprIR::constExprIt loop_end_pos,
+                                 size_t loop_depth, size_t vector_size, const std::vector<LoweredExprPtr>& body_exprs) {
+    // Note: currently we simply take out td of the last expr in the loop. If needed,
+    // this can be generalized for loops with multiple different out td's.
+    const auto& out_td = std::prev(loop_end_pos)->get()->get_outputs().front();
```

Comment:
Thoughts: it's a very slippery path because there might be several `out_tds` with different tensor shapes. In fact, I had to fix something like this recently in the LIR PR. But the good news is that we have exit and entry points now! So we can use them to derive our work amount correctly. No saying we need to fix it right now, but it would get us into trouble eventually.

Reply:
Yeah, it's valid since we mark simple loops with several inputs and outputs without loop fusion.
Added support of several tensors. Thanks!